pkg/api/persistentvolumeclaim/util.go

```diff
@@ -33,6 +34,60 @@ func DropDisabledFields(pvcSpec, oldPVCSpec *core.PersistentVolumeClaimSpec) {
 	if !dataSourceIsEnabled(pvcSpec) && !dataSourceInUse(oldPVCSpec) {
 		pvcSpec.DataSource = nil
 	}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) && !allocatedResourcesInUse(oldPVCSpec) {
+		pvcSpec.AllocatedResources = nil
+	}
+}
+
+// SetAllocatedResources ensures that AllocatedResources field can only increase and tracks
+// maximum user requested capacity.
+func SetAllocatedResources(pvcSpec, oldPVCSpec *core.PersistentVolumeClaimSpec) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) {
+		userResources := pvcSpec.Resources
+
+		// for new PVC creation we will simply set it to user requested size
+		// even if user provided one.
+		if oldPVCSpec == nil {
+			pvcSpec.AllocatedResources = &userResources
```

Comment:
looking at the code, I do not think it is possible to prevent users from setting this field altogether. This is because validation is called *after* `PrepareForCreate` or `PrepareForUpdate` and if we stopped this field from being set in validation then we won't be able to set this field in `PrepareForCreate` or `PrepareForUpdate` as well.

Reply:
The only thing we *can* do is - if user does set this field then silently drop and replace it with defaults selected from `SetAllocatedResources`

Reply:
Ok